src/time.ts

```diff
@@ -88,7 +88,10 @@ export const stringify12Hour = (time: NetworkTime) => {
     return `${resolvedHours} ${isPM ? "PM" : "AM"}`;
 };
 
-export const stringifyDuration = (duration: Duration, useFull = false) => {
+export const stringifyDuration = (duration: Duration | null, useFull = false) => {
+    if (duration === null) {
+        return "---";
```

Comment:
This is pretty localized to a specific component, right? I think I'd rather keep that fallback there, and continue to enforce that this function must take a non-null `Duration`.